src/core/frameworks/detect.ts

```diff
@@ -0,0 +1,346 @@
+import { promises as fs } from "fs";
+import globrex from "globrex";
+import path from "path";
+import { DEFAULT_CONFIG } from "../../config";
+import { logger, stripJsonComments } from "../utils";
+import { apiFrameworks, appFrameworks } from "./frameworks";
+
+const packageJsonFile = "package.json";
+
+export async function generateConfiguration(app?: DetectedFolder, api?: DetectedFolder): Promise<FrameworkConfig> {
+  let config: FrameworkConfig = {
+    appLocation: DEFAULT_CONFIG.appLocation!,
+    outputLocation: DEFAULT_CONFIG.outputLocation!
+  };
+
+  if (!app && !api) {
+    logger.silly("No known frameworks detected");
+    return config;
+  }
+
+  let name = '';
+
+  if (app) {
+    name += `${app.frameworks.map(f => f.name).join(', ')}`;
+    app.frameworks.forEach(f => config = { ...config, ...f.config });
+    config.appLocation = await computePath(app.rootPath, config.appLocation);
+    config.outputLocation = await computePath(config.appLocation, config.outputLocation);
+  }
+
+  if (api) {
+    name += (name ? ', with ' : 'No app frameworks detected, ');
+    name += `API: ${api.frameworks.map(f => f.name).join(', ')}`;
+    api.frameworks.forEach(f => config = { ...config, ...f.config });
+
+    // If api is not under app folder, we must find a common root folder
+    if (app && !isDescendantPath(api.rootPath, config.appLocation!)) {
+      const movePath = path.relative(config.appLocation!, app.rootPath);
+      console.log({ movePath });
+      
+      // TODO: would be better to lowest common root folder instead
+      config.appLocation = '.';
+      console.log(config.outputLocation);
+
+      logger.silly(`Api folder is not under app folder, using common root: ${config.appLocation}`);
+
+      config.outputLocation = path.normalize(path.join(movePath, config.outputLocation!));
+    }
+
+    const apiRootPath = path.normalize(path.relative(config.appLocation!, api.rootPath));
+    config.apiLocation = await computePath(apiRootPath, config.apiLocation);
+  }
+
+  config.name = name;
+  return config;
+}
+
+async function computePath(basePath: string, additionalPath?: string): Promise<string> {
+  if (!additionalPath) {
+    return basePath;
+  }
+
+  if (!additionalPath.startsWith('{')) {
+    return path.join(basePath, additionalPath);
+  }
+  
+  const match = additionalPath.match(/^\{(.*?)#(.*?)\}$/);
+  const [, filename, expression] = match || [];
+  if (!filename || !expression) {
+    throw new Error(`Invalid dynamic path format: ${additionalPath}`);
+  }
+
+  const files = await getFiles(basePath);
+  const file = findFile(filename, files);
+  if (!file) {
+    throw new Error(`File "${filename}" not found in dynamic path: ${additionalPath}`);
+  }
+
+  const json = await safeReadJson(file);
+  if (!json) {
+    throw new Error(`Invalid JSON file: ${file}`);
+  }
+
+  const evaluateExpression = (json: JsonData, expr: string) =>
+    Function(`"use strict";return data => (${expr})`)()(json);
```

Comment:
I am afraid this will fail CodeQL. Let's wait and see if the final build passes.